branchLayoutApi/src/main/java/com/virtuslab/branchlayout/api/BranchLayout.java

```diff
@@ -51,4 +54,49 @@ public IBranchLayout slideOut(String branchName) throws BranchLayoutException {
       return List.of(entry.withChildren(children.flatMap(child -> slideOut(child, entryToSlideOut))));
     }
   }
+
+  @Override
+  public IBranchLayout slideIn(String aboveBranchName, String branchName) throws BranchLayoutException {
+    var entry = findEntryByName(branchName).getOrNull();
+    if (entry != null) {
+      throw new BranchLayoutException("Child branch entry '${branchName}' already exists");
+    }
+    var parentEntry = findEntryByName(aboveBranchName).getOrNull();
+    if (parentEntry == null) {
+      throw new BranchLayoutException("Parent branch entry '${aboveBranchName}' does not exists");
+    }
+    var entryToSlideIn = new SlidInEntry(branchName, null, List.empty());
+    return new BranchLayout(rootEntries.flatMap(rootEntry -> slideIn(rootEntry, entryToSlideIn, parentEntry)));
+  }
+
+  @SuppressWarnings("interning:not.interned") // to allow for `entry == entryToSlideOut`
+  private List<IBranchLayoutEntry> slideIn(
+      IBranchLayoutEntry entry,
+      IBranchLayoutEntry entryToSlideIn,
+      IBranchLayoutEntry parent) {
```

Comment:
Hmm consider exposing `getParent` from the `BranchLayoutEntry`... just as done for `GitMacheteNonRootBranch`, or on #415 for `MyBranchLayoutEntry`

Reply:
Actually I see no use for `getParent` in my case. The one passed here is the very same entry that has been found before within `IBranchLayout slideIn(String parentBranchName, String newBranchName)`.

Reply:
issue created https://github.com/VirtusLab/git-machete-intellij-plugin/issues/430